tests/util/rest.go

```diff
@@ -52,3 +55,47 @@ func FormFromFile(t *testing.T, form map[string]string) (*bytes.Buffer, string)
 
 	return body, mp.FormDataContentType()
 }
+
+func RetrievePipelineId(t *testing.T, APIServerURL string) string {
+	response, err := http.Get(fmt.Sprintf("%s/apis/v2beta1/pipelines", APIServerURL))
+	require.NoError(t, err)
+	responseData, err := io.ReadAll(response.Body)
+	require.NoError(t, err)
+	type Pipeline struct {
+		Pipelines []struct {
+			PipelineID  string `json:"pipeline_id"`
+			DisplayName string `json:"display_name"`
+		} `json:"pipelines"`
+	}
+	var pipelineData Pipeline
+	var pipelineID string
+	err = json.Unmarshal(responseData, &pipelineData)
+	require.NoError(t, err)
+	for _, pipeline := range pipelineData.Pipelines {
+		if pipeline.DisplayName == "[Demo] iris-training" {
+			pipelineID = pipeline.PipelineID
+			break
+		}
+	}
+	return pipelineID
+}
+
+func FormatRequestBody(t *testing.T, pipelineID string) []byte {
+	type PipelineRequest struct {
+		DisplayName              string `json:"display_name"`
+		PipelineVersionReference struct {
+			PipelineID string `json:"pipeline_id"`
+		} `json:"pipeline_version_reference"`
+	}
```

Comment:
let's pull this out of the function so we can re-use it for other pipeline tests in the future